### DISCUSSION TITLE
Feature: Add deep linking in case of no data result

### DISCUSSION THREAD

**Discussion by papagian** [grafana/grafana#41744] [Data Sources]
**What would you like to be added**:
Cloud datasources having deep linking for directing the users from grafana panels to the respective cloud. Currently this is provided only if there are some data returned ([cloud monitoring](https://github.com/grafana/grafana/blob/1dd5b58e957625a18e44242af3c5acb315609a2f/public/app/plugins/datasource/cloud-monitoring/datasource.ts#L49) and [azure](https://github.com/grafana/grafana/blob/f0726cc6ad6f20d1e7f188961b504433678b6382/public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts#L156)).
However, such a link would be also useful if no data are returned.

An additional challenge for cloud monitoring is that we [look for](https://github.com/grafana/grafana/pull/25858#issuecomment-651682240) the resource type information for a given metric type (if not provided) in the Monitoring API response so if there isn't be such a response (and the resource type is not included in the query filter) then we won't have a way to get that information.

**Comment by bergquist** [comment]
@kylebrandt @ryantxu Any thoughts about returning deep links in dataframes as reserved fields? or something similar.

**Comment by kylebrandt** [comment]
I'm not sure I fully understand the situation. If there is no data ... what exactly do we want to show where? Also, is a "Deep link" in someway different than a "data link"?

**Comment by kylebrandt** [comment]
Presently datalinks are associated with a Field (https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go/data?tab=doc#FieldConfig see Links property). So one could return fields of zero length with data links, but I'm not sure the UI would provide a place to click them?

**Comment by eraac** [comment]
I think is a desired behavior (at least for Stackdriver) https://github.com/grafana/grafana/pull/26737 
This PR was done after a discussion in slack https://grafana.slack.com/archives/C0Y4TLW74/p1596095359394500

From what I understood, the deep link require a value that only timeseries data returns

**Comment by sunker** [comment]
@papagian was there a specific request for this? The resource type was removed from the deep link (if not specified by the user in the grafana gui) in [this](https://github.com/grafana/grafana/pull/26551) pr.

**Comment by papagian** [comment]
@sunker there was no a specific request but more a nice to have behaviour that I realised during the implementation. 
 The difficulty for me was that in the graph (frontend) we used to add the link in the returned time series but how to keep the link when there are no data returned? Dropping the requirement for the resource type makes things a bit easier.

**Comment by sunker** [comment]
I see, thanks!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
